app/src/main/java/com/paulacr/breakfastplaces/MainActivity.kt

```diff
@@ -3,41 +3,63 @@ package com.paulacr.breakfastplaces
 import android.os.Bundle
 import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
+import androidx.compose.foundation.layout.Column
 import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.material.Button
 import androidx.compose.material.MaterialTheme
 import androidx.compose.material.Surface
 import androidx.compose.material.Text
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.tooling.preview.Preview
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.rememberNavController
+import com.paulacr.breakfastplaces.photo.Picture
 import com.paulacr.breakfastplaces.ui.theme.BreakfastPlacesTheme
 
 class MainActivity : ComponentActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContent {
-            BreakfastPlacesTheme {
-                // A surface container using the 'background' color from the theme
-                Surface(
-                    modifier = Modifier.fillMaxSize(),
-                    color = MaterialTheme.colors.background
-                ) {
-                    Greeting("Android")
-                }
+            val navController = rememberNavController()
+            NavHost(navController = navController, startDestination = "start") {
+                composable("start") { Extracted { navController.navigate("picture") } }
+                composable("picture") { Picture() }
+            }
+        }
+    }
+
+    @Composable
+    private fun Extracted(openCamera: () -> Unit) {
+        BreakfastPlacesTheme {
+            // A surface container using the 'background' color from the theme
+            Surface(
+                modifier = Modifier.fillMaxSize(),
+                color = MaterialTheme.colors.background
+            ) {
+                Greeting("Android", openCamera)
             }
         }
     }
 }
 
 @Composable
-fun Greeting(name: String) {
-    Text(text = "Hello $name!")
+fun Greeting(name: String, openCamera: () -> Unit) {
```

Comment:
let's rename to OpenCameraScreen?

Reply:
a356a6f961c44ad888314ce77fa37160be05441d